### DISCUSSION TITLE
How to use Docker Sqitch with CI/CD?

### DISCUSSION THREAD

**Discussion by hrkachhia** [theory/sqitch#759] [Q&A]
I want to use Docker Sqitch in my production environment. It is working fine on my machine. Now, I want to use Docker Sqitch on AWS. Do you have any ideas on how to achieve this? I have tried using AWS ECR, but it is not working. Can you please provide guidance?

**Comment by theory** [comment]
You can pull from Docker Hub from within AWS, no need for ECR.

**Reply by hrkachhia** [reply]
Do you have any instructional guidelines? If so, please provide them to me.

As mentioned in the Docker Hub overview page, am I required to rebuild the Docker image for Snowflake? What steps should I take after pulling the image from Docker Hub? Will it work directly?

**Reply by theory** [reply]
Ah, you hadn't mentioned Snowflake before. For that you'll have to install the dependencies inside the core Docker image. See #736 for some relevant discussion.

**Reply by hrkachhia** [reply]
I have read the discussion which you have provided to me.
I have get the file snowflake.sh script for install sqitch and snowflake dependencies.
I have move snowflake.sh file inside the snowflake folder and added below line in snowflake docker file.

COPY snowflake.sh ./snowflake.sh
Run chmod +x /snowflake.sh
cmd ["./snowflake.sh"]

But still, I am getting error [unixODBC][Driver Manager]Data source name not found, and no default driver specified (SQL-IM002)

Can you please guide me to resolve this issue.

**Reply by hrkachhia** [reply]
I have successfully built docker sqitch image push it on the AWS ECR. I have check the log when built. It has installed all the dependencies. After that I have set env variable SQITCH_IMAGE with my ECR repo path and I have also add repo path in docker-sqitch.sh file. but when I try to execute command docker-switch.sh verify it is giving me error :

[unixODBC][Driver Manager]Can't open lib '/usr/lib/snowflake/odbc/lib/libSnowflake.so' : file not found (SQL-01000)

Can you please guide me  to resolve this error.

**Reply by theory** [reply]
Looks like you need to build the Snowflake image, not just the base Sqitch image. See the [Snowflake README](/sqitchers/docker-sqitch/tree/main/snowflake) for details.

**Reply by hrkachhia** [reply]
I have already built the snowflake image for the snowflake and push it on AWS ECR. Then after I am getting this error when I am trying to use that image.

**Reply by theory** [reply]
That error indicates that there is no Snowflake ODBC driver, which suggests that you're not actually creating the Snowflake image.

**Reply by hrkachhia** [reply]
Maybe this issue is occur because of I have built snowflake docker image on codebuild. 
I have pushed working snowflake docker image from my Linux machine to AWS ECR and the after I have tried to use. 
For connection to snowflake, I have declared all the require snowsql environment variables as below: 

- SNOWSQL_ACCOUNT=snowflake account
- export SNOWSQL_USER=snowflake user
- export SQITCH_PASSWORD=my snowflake password 
- export SNOWSQL_ROLE=Admin
- export SNOWSQL_WAREHOUSE=COMPUTE_WH

Now, I am getting error related to privileges when I tried to deploy from AWS codebuild. 


003001 (42501): SQL access control error:
Insufficient privileges to operate on database 'test_db'
"snowsql" unexpectedly returned exit value 5

I am able to perform sqitch deploy ,verify and revert from my linux machine from same user and role but getting error only in AWS codebuild.

**Reply by theory** [reply]
Did you mean to export `SNOWSQL_ACCOUNT`?

**Reply by hrkachhia** [reply]
@theory : I have applied some changes in docker-switch.sh file. After those changes it is working fine with AWS ECR. I want to share those changes and contribute in this project. Can you please allow me to create pull request for it?

**Reply by theory** [reply]
You can fork the project and create a PR from there. Or just drop the patch here if you'd rather.
